src/alpaka_job_coverage/validate.py

```diff
@@ -0,0 +1,388 @@
+
+    Returns:
+        Dict[str, Tuple[str, str]]: Returns parsed compiler name and version
+    """
+    validated_args = {}
+    known_compiler = [GCC, CLANG, NVCC, CLANG_CUDA, HIPCC]
+
+    for parameter_name, parameter in arguments.items():
+        if "@" not in parameter:
+            exit_error(f"@ is missing in {parameter_name}={parameter}")
+
+        splitted_name_version = parameter.split("@", 1)
+        name = splitted_name_version[0]
+        version = splitted_name_version[1]
+
+        if name not in known_compiler:
+            exit_error(f"Unknown compiler: {name}\nKnown Compiler: {known_compiler}")
+
+        # use parse() function to validate that the version has a valid shape
+        try:
+            pk_version.parse(version)
+        except Exception as e:
+            exit_error(f"Could not parse version number of {name}: {version}")
+
+        validated_args[parameter_name] = (name, version)
+
+    return validated_args
+
+
+@typechecked
+def validate_args_backend(arguments: List[str]) -> Dict[str, List[Tuple[str, str]]]:
+    """Check if backend names exist and version numbers are parsable.
+
+    Args:
+        arguments (List[str]): List of Backend names with versions.
+
+    Returns:
+        Dict[str, List[Tuple[str, str]]]: List of parsed backend names with
+        versions.
+    """
+    validated_args = {BACKENDS: []}
+
+    for backend in arguments:
+        if backend not in BACKENDS_LIST:
+            if "@" not in backend:
+                exit_error(f"@ is missing in {backend}")
+
+            splitted_name_version = backend.split("@", 1)
+            name = splitted_name_version[0]
+            version = splitted_name_version[1]
+
+            if name not in BACKENDS_LIST:
+                exit_error(f"Unknown backend: {name}\nKnown Compiler: {BACKENDS_LIST}")
+
+            # use parse() function to validate that the version has a valid shape
+            try:
+                pk_version.parse(version)
+            except Exception as e:
+                exit_error(f"Could not parse version number of {name}: {version}")
+
+            validated_args[BACKENDS].append((name, version))
+
+    return validated_args
+
+
+@typechecked
+class VersionAction(argparse.Action):
+    # check if argument has valid version shape
+    def __call__(self, parser, namespace, values, option_string):
+        try:
+            pk_version.parse(values)
+        except Exception as e:
+            print(cs(f"ERROR: wrong version for {option_string}", "Red"))
+            raise e
+        setattr(namespace, self.dest, values)
+
+
+@typechecked
+def validate_version_support(
+    parameters: Dict[str, Union[Tuple[str, str], List[Tuple[str, str]]]],
+):
+    """Print a warning, if a software version is not official supported by the
+    library. The function does not exit the application, because the versions
+    can still be tested with the filter functions.
```

Comment:
```suggestion
    """Prints a warning if a software version is not officially supported by the
    library. This function does not exit the application because the versions
    can still be tested with the filter functions.
```